code/API_definitions/Edge-Application-Management.yaml

```diff
@@ -961,6 +961,231 @@ components:
+              example: cilium
+            version:
+              description: CNI provider version
+              type: string
+              example: "1.13"
+        additionalNetworks:
+          description: Additional Networks for the Kubernetes cluster.
+          type: array
+          items:
+            type: object
+            description: Additional network interface definition
+            properties:
+              name:
+                description: Additional Network Name
+                type: string
+                example: net1
+              interfaceType:
+                description: |
+                  Type of additional Interface:
+                  netdevice: (SR-IOV) A regular kernel network device in the
+                             Network Namespace (netns) of the container
+                  vfio-pci: (SR-IOV) A PCI network interface directly mounted
+                            in the container
+                  interface: Additional interface to be used by cni plugins
+                             such as macvlan, ipvlan
+                  Note: The use of SR-IOV interfaces automatically
+                  configure the required kernel parameters for the nodes.
+                type: string
+                example: vfio-pci
+                enum:
+                - netdevice
+                - vfio-pci
+                - interface
+
+    AdditionalStorage:
+      description:  Additional storage for the application.
+      type: array
+      items:
+        type: object
+        required:
+          - storageSize
+          - mountPoint
+        properties:
+          name:
+            type: string
+            description: Name of additional storage resource.
+            example: logs
+          storageSize:
+            type: string
+            description: Additional persistent volume for the application.
+            example: 80GB
+            pattern: ^\d+(GB|MB)$
+          mountPoint:
+            type: string
+            description: Location of additional storage resource.
+            example: /logs
+
+    Vcpu:
+      type: string
+      pattern: ^\d+((\.\d{1,3})|(m))?$
+      description: |
+        Number of vcpus in whole (i.e 1), decimal (i.e 0.500) up to
+        millivcpu, or millivcpu (i.e 500m) format.
+      example: "500m"
+
+    Kubernetes:
+      description: Definition of Kubernetes Cluster Infrastructure.
+      type: object
+      required:
+        - nodePools
+        - infraKind
+      properties:
+        infraKind:
+          description: Type of infrastructure for the application.
+          type: string
+          example: kubernetes
+          enum:
+            - kubernetes
+        version:
+          type: string
+          description: Minimum Kubernetes Version.
+          example: "1.29"
+        controlNodes:
+          type: integer
+          description: Number of nodes for Kubernetes control plane.
+          enum:
+          - 1
+          - 3
```

Comment:
Not sure why an enum is defined for the controlNodes integer, shouldn't it allow any integer greater than 0?

Reply:
Sorry, I realize now this is for master nodes, so ignore my previous comment. [Docs](https://kubernetes.io/docs/tasks/administer-cluster/configure-upgrade-etcd/#multi-node-etcd-cluster) recommend up to 5 nodes for large clusters, I don't think we'll be dealing with large clusters here but perhaps for completeness add an enum value for 5.

Reply:
Okey, the idea here was to offer Control Plane w/HA and wo/HA but at the end it is controlled by the operator. I think better approach is a boolean controlPlaneHa: true/false

Reply:
I'm not sure if this should be part of the API. Application provider should have SLA in place with the operators and they shouldn't care about how the control plane of the operator infra is implemented.